lua/ttt2/libraries/events.lua

```diff
@@ -0,0 +1,350 @@
+
+		deprecatedEvents[#deprecatedEvents + 1] = deprecatedEventData
+	end
+
+	return deprecatedEvents
+end
+
+if SERVER then
+	---
+	-- Triggers an event, adds it to a managed list and starts the score calculation for this event.
+	-- @param string name The name identifer of the event
+	-- @param varag The arguments that should be passed to the event, see the @{EVENT:Trigger} function
+	-- @realm server
+	function events.Trigger(name, ...)
+		if not events.Exist(name) then
+			ErrorNoHalt("[TTT2] ERROR: An event with the name '" .. tostring(name) .. "' does not exist.\n")
+
+			return
+		end
+
+		local newEvent = events.Create(name)
+
+		-- only add new event to managed event list, if addition was not aborted
+		if newEvent:Trigger(...) then
+			events.list[#events.list + 1] = newEvent
+		else
+			return
+		end
+
+		-- add to deprecated score list
+		local deprecatedEventData = newEvent:GetDeprecatedFormat(newEvent.event)
+
+		if deprecatedEventData then
+			SCORE:AddEvent(deprecatedEventData)
+		end
+
+		-- run a hook with the newly added event
+		hook.Run("TTT2AddedEvent", name, newEvent)
+	end
+
+	---
+	-- Streams the whole event table to all clients, usually done after the round ended.
+	-- @internal
+	-- @realm server
+	function events.StreamToClients()
+		local eventList = events.list
+		local eventStreamData = {}
+
+		for i = 1, #eventList do
+			eventStreamData[i] = eventList[i]:GetNetworkedData()
+		end
+
+		net.SendStream("TTT2_EventReport", eventStreamData)
+	end
+
+	---
+	-- This function is called in @{GM:TTTEndRound} and updates the scores and kills
+	-- of every player in the scoreboard.
+	-- @internal
+	-- @realm server
+	function events.UpdateScoreboard()
+		local scores = events.GetTotalPlayerScores()
+		local deaths = events.GetTotalPlayerDeaths()
+
+		for ply64, score in pairs(scores) do
+			ply = player.GetBySteamID64(ply64)
+
+			if not IsValid(ply) or not ply:ShouldScore() then continue end
+
+			ply:AddFrags(score)
+		end
+
+		for ply64, death in pairs(deaths) do
+			ply = player.GetBySteamID64(ply64)
+
+			if not IsValid(ply) or not ply:ShouldScore() then continue end
+
+			ply:AddDeaths(death)
+		end
+	end
+
+	---
+	-- This hook is called once an event occured, the data is processed
+	-- and it is about to be added. This hook can be used to modify the data
+	-- or to cancel the event by returning `false`.
+	-- @param string type The type of the event as in `EVENT_XXX`
+	-- @param table eventData The table with the event data
+	-- @return boolean Return false to cancel the addition of this event
+	-- @realm server
+	-- @hook
+	function GM:TTT2OnTriggeredEvent(type, eventData)
+
+	end
+
+	---
+	-- This hook is called after the event was successfully added to the
+	-- eventmanager.
+	-- @param string type The type of the event as in `EVENT_XXX`
+	-- @param EVENT event The event that was added with all its functions
+	-- @realm server
+	-- @hook
+	function GM:TTT2AddedEvent(type, event)
+
+	end
+end
+
+if CLIENT then
```

Comment:
```suggestion
else -- CLIENT
```

Reply:
I really dislike `else -- CLIENT` because there is no benefit to it and it looks ugly in my opinion.

Reply:
Yea theortically the content has to be moved to it's clientside / serverside files